File: app/src/main/java/com/duy/interpreter/exceptions/UnrecognizedTokenException.java
package com.duy.interpreter.exceptions;

import com.duy.interpreter.tokens.Token;

public class UnrecognizedTokenException extends com.duy.interpreter.exceptions.ParsingException {
    public Token token;

    public UnrecognizedTokenException(Token t) {
        super(t.lineInfo, "The following token doesn't belong here: " + t);
        this.token = t;
    }
}

File: libCompiler/src/main/java/com/duy/pascal/backend/parse_exception/UnrecognizedTokenException.kt
package com.duy.pascal.backend.parse_exception

import com.duy.pascal.backend.tokens.Token

class UnrecognizedTokenException : ParsingException {
    var token: Token

    constructor(token: Token) : super(token.lineNumber, "The following name doesn't belong here: $token") {
        this.token = token
    }
}
